fip-6.md

```diff
@@ -0,0 +1,376 @@
+|max_fee|Yes|max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|actor|Yes|FIO account name|FIO account for the signer|
+###### Example
+```
+{
+  "lock_id": 1,
+  "actor": "aftyershcu22",
+  "amount": 400000000000
+  "max_fee": 40000000000
+}
+```
+##### Processing
+* require auth of the actor
+* Verify format and values for lock_id, amount, max_fee, actor. invalid input error if incorrect.
+* Verify the lock id exists in the locktokens table, error if none exist.
+* Verify the owner of the lock matches the actor. signature error if no match
+* perform the locking schedule for the lock (updating record of unlocked tokens)
+* Verify the lock in question has a remaining_locked_amount >= amount else (insufficient locked funds for unlocking error)
+* update the remaining locked amount -= amount
+* verify tx does not exceed max transaction size.
+* increase RAM limit by 200+(60*number of lock periods) bytes
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid lock id|Invalid format|400|"lock_id"|lock id invalid|"Invalid Lock ID"|
+|Invalid Actor|Actor does not exist on chain|403|||"Invalid Actor"|
+|Insufficient lock balance| Lockdoes not have required amount of funds|400|"amount"|Value sent in, e.g. "100"|"Insufficient lock balance"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||fee_collected|String|fee amount collected SUFs|
+###### Example
+```
+{
+  "status": "Ok",
+  "fee_collected": 0        
+}
+```
+## Fees
+a new fee will be created for lock_tokens type 2, 600000000 SUF per lock period, this is a mandatory fee.
+
+
+## Specification
+### Clean lock table
+#### New end point: *clean_lock_table* 
+#### New action in new fio.lock contract cleanlocktbl
+, 
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+
+
+##### Processing
+* require auth fio system accounts.
+* read the locktokens table by remaining_locked_amount
+* if first entry has remaining locked amount > 0 throw no work exception.
+* traverse list until remaining locked amount > 0 or 50 entries removed.
+* remove entry with remaining locked amound == 0
+* return status ok and removed_count
+
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|No work|No items to remove.|403|||"No work"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||removed_count|String|number of items removed|
+###### Example
+```
+{
+   "status": "Ok",
+   "removed_count": "25"        
+}
+```
+
+## Specification
+### Clean locks
```

Comment:
This appears the same as clean_lock_table. What's the difference?

Reply:
no difference in the return.

Reply:
Where is the difference?

Reply:
a description the the differences is added to the FIP. a description of each request at the high level.